### PR TITLE
Add unit tests for active_deadline.go

### PR DESCRIPTION
**Pull request by loktev-d**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Improves code coverage for kubelet

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by loktev-d** [comment]
@dchen1107 @mrunalp please take a look

**Comment by pacoxu** [comment]
This will not fix #109717, and it is only part of it, right. Please edit the pull request description.
/ok-to-test

**Comment by loktev-d** [comment]
@pacoxu Done

File: pkg/kubelet/active_deadline_test.go [hunk 1]
```diff
@@ -42,9 +46,51 @@ func (m *mockPodStatusProvider) GetPodStatus(uid types.UID) (v1.PodStatus, bool)
...
+		{nil, fakeRecorder, fakeClock},
+		{nil, fakeRecorder, nil},
+		{nil, nil, fakeClock},
+		{nil, nil, nil},
+	}
+
+	for i, testCase := range testCases {
+		actual, err := newActiveDeadlineHandler(testCase.podStatusProvider, testCase.recorder, testCase.clock)
+
+		if i == 0 {
```
Comment (rphillips) [inline pkg/kubelet/active_deadline_test.go:73]: This appears to only be testing if we can construct an ActiveDeadlineHandler object... Am I missing something?

Comment (loktev-d) [reply pkg/kubelet/active_deadline_test.go:73]: Yep, there is actually only line 54 (which returns error) needs to be covered, but I think that unit test of function NewActiveDeadlineHeader won't be complete without no errors scenario, and I also saw such testing case in other tests

Comment (haircommander) [reply pkg/kubelet/active_deadline_test.go:73]: I think it's worth adding a comment about that. Something like `0th case is the only one expected to pass, and is kept for coverage and confidence check`

Comment (loktev-d) [reply pkg/kubelet/active_deadline_test.go:74]: @haircommander Done

**Review by mrunalp — APPROVED** [review]

**Pull request merged**